src/absil/ilwrite.fs

```diff
@@ -3916,15 +3936,7 @@ let writeBinaryAndReportMappings (outfile, ilg, pdbfile: string option, signer:
 
     let timestamp = absilWriteGetTimeStamp ()
 
-    let os = 
-        try  
-            new BinaryWriter(FileSystem.FileStreamCreateShim(outfile))
-        with e -> 
-            failwith ("Could not open file for writing (binary mode): " + outfile)    
-
-    let  pdbData,debugDirectoryChunk,debugDataChunk,textV2P,mappings =
-        try 
-      
+    let gen os (pdbfile: bool) =
```

Comment:
ok , the `gen` function is like that after several refactoring attempts.
I'd have liked to move this function outside , but the diff was too big, so I left it like that.

I always use inner function because mvoe value binding later, so i can easy see values captured inside big functions like this one.

I will do as you said, less diff.